### PR TITLE
UGC-18860: Create short video template

### PR DESCRIPTION
**Pull request by andy-pei**
## Description
Create shortvideo template to only play short videos from:
* Instagram
* Youtube
* Tiktok

## Checklist

- [ ] Pull Request is properly described.
- [ ] The corresponding Jira ticket is updated.
- [ ] I have requested a review from at least one reviewer.
- [ ] The changes are tested
- [ ] I have verified my UX changes with a designer
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: widgets/shortvideo/_inline-tile.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          overflow: hidden;
+          border-radius: dimens.$inline-tile-border-radius;
+          position: relative;
+          opacity: 1;
+        }
+      }
+    }
+  }
+
+  .icon-play {
```
Comment (manikandan-ravikumar) [inline widgets/shortvideo/_inline-tile.scss:35]: Shall we use the icon from the global `icon.scss`?

File: widgets/shortvideo/_inline-tile.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    height: dimens.$play-icon_height;
+    background-image: url('data:image/svg+xml;charset=UTF-8,<svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" fill="none"><circle cx="15" cy="15" r="15" fill="%23BCBBBC"/><path d="M19.5 14.1336C20.1667 14.5185 20.1667 15.4808 19.5 15.8657L13.5 19.3298C12.8333 19.7147 12 19.2335 12 18.4637L12 11.5355C12 10.7657 12.8333 10.2846 13.5 10.6695L19.5 14.1336Z" fill="black"/></svg>');
+    background-size: contain;
+  }
+
+  & ~ .swiper-button-next,
+  & ~ .swiper-button-prev {
+    --swiper-navigation-sides-offset: 0px;
+
+    top: calc(var(--swiper-navigation-top-offset, 50%) - dimens.$nosto-container-margin + 10) !important;
```
Comment (manikandan-ravikumar) [inline widgets/shortvideo/_inline-tile.scss:46]: Is the `!important` flag necessary?

File: widgets/shortvideo/inline-shortvideo-swiper.loader.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
+import { Sdk, SwiperData } from "types"
```
Comment (manikandan-ravikumar) [inline widgets/shortvideo/inline-shortvideo-swiper.loader.ts:1]: How is this different from carousel swiper initialization? If the only different is the id and navigation button parameters, should we somehow reuse the logic from carousel? Maybe we can work on this later in a separate PR

File: package.json [hunk 4]
```diff
@@ -88,7 +88,7 @@
   "dependencies": {
     "@appnest/masonry-layout": "^2.2.3",
     "@glidejs/glide": "^3.6.1",
-    "@stackla/handlebars": "^0.1.3",
+    "@stackla/handlebars": "^0.1.5",
```
Comment (manikandan-ravikumar) [inline package.json:91]: Did we introduce any hbs helpers specific to rendering short videos?

Comment (andy-pei) [reply package.json:91]: Yes @manikandan-ravikumar , I created a new ifShortVideo hbs helper for that, you can view the details here https://github.com/Stackla/stackla-packages/pull/123/files

File: widgets/shortvideo/_inline-tile.scss (hunk 1 above)
Comment (andy-pei) [reply widgets/shortvideo/_inline-tile.scss:35]: This icon is not needed in shortvideo, I will remove it

File: widgets/shortvideo/_inline-tile.scss (hunk 2 above)
Comment (andy-pei) [reply widgets/shortvideo/_inline-tile.scss:46]: This is not needed, thanks for pick this up

File: widgets/shortvideo/inline-shortvideo-swiper.loader.ts (hunk 3 above)
Comment (andy-pei) [reply widgets/shortvideo/inline-shortvideo-swiper.loader.ts:1]: It is the same as carousel swiper, we can abstract the swiper and reuse it. Let's do it in a separate PR

File: widgets/shortvideo/inline-shortvideo-swiper.loader.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  const screenSize = window.innerWidth
+  const perView = !enable_custom_tiles_per_page
+    ? Math.floor(screenSize / (tileWidth + 10))
+    : // FIXME: All numbers should be numbers across the board
+      parseInt(tiles_per_page)
+
+  sdk.tiles.setVisibleTilesCount(perView * 2)
+
+  initializeSwiper({
+    id: "inline-shortvideo",
```
[the remark this answers is not in the document]
Comment (andy-pei) [reply widgets/shortvideo/inline-shortvideo-swiper.loader.ts:43]: Not quite sure about this one. I need more understanding of the Swiper on Carousel. Probably we can fix it when we work on the separate PR where we reuse it across different templates

**Review by manikandan-ravikumar — APPROVED** [review]

**Pull request merged**